### PR TITLE
Reduce the result set to only include what needs to be send

### PR DESCRIPTION
**Pull request by kristianschneider**
Right now we send all steps to the coordinator.
This is not good because we also collect the data frames for each step and thereby vastly reduce performance

### CODE REVIEW

**Review by ZavezX — CHANGES_REQUESTED** [review]
Many of the results that are left out of the result set needs to be saved and sent to the coordinator. But I think it is a good idea to leave the ones out of the result set, that we don't need.

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 1]
```diff
@@ -87,86 +86,86 @@
...
-results['hourly_settled_consumption_ga_brp'] = aggregate_per_ga_and_brp(results['hourly_consumption_df'])
+hourly_settled_consumption_ga_brp = aggregate_per_ga_and_brp(results['hourly_consumption_df'])
 
 # STEP 17
-results['flex_settled_consumption_ga_brp'] = aggregate_per_ga_and_brp(results['flex_consumption_with_grid_loss'])
+flex_settled_consumption_ga_brp = aggregate_per_ga_and_brp(flex_consumption_with_grid_loss)
 
 # STEP 18
-results['hourly_production_ga'] = aggregate_per_ga(results['hourly_production_with_system_correction_and_grid_loss'])
+hourly_production_ga = aggregate_per_ga(results['hourly_production_with_system_correction_and_grid_loss'])
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:154]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102180 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 2]
```diff
@@ -87,86 +86,86 @@
...
 
 # STEP 19
-results['hourly_settled_consumption_ga'] = aggregate_per_ga(results['hourly_consumption_df'])
+hourly_settled_consumption_ga = aggregate_per_ga(results['hourly_consumption_df'])
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:157]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102180 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 3]
```diff
@@ -87,86 +86,86 @@
...
 
 # STEP 20
-results['flex_settled_consumption_ga'] = aggregate_per_ga(results['flex_consumption_with_grid_loss'])
+flex_settled_consumption_ga = aggregate_per_ga(flex_consumption_with_grid_loss)
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:160]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102180 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 4]
```diff
@@ -87,86 +86,86 @@
...
 
 # STEP 21
-results['total_consumption'] = calculate_total_consumption(results['net_exchange_per_ga_df'], results['hourly_production_ga'])
+total_consumption = calculate_total_consumption(results['net_exchange_per_ga_df'], hourly_production_ga)
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:163]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102180 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 5]
```diff
@@ -87,86 +86,86 @@
...
-results['hourly_settled_consumption_ga_es'] = aggregate_per_ga_and_es(results['hourly_consumption_df'])
+hourly_settled_consumption_ga_es = aggregate_per_ga_and_es(results['hourly_consumption_df'])
 
 # STEP 14
-results['flex_settled_consumption_ga_es'] = aggregate_per_ga_and_es(results['flex_consumption_with_grid_loss'])
+flex_settled_consumption_ga_es = aggregate_per_ga_and_es(flex_consumption_with_grid_loss)
 
 # STEP 15
-results['hourly_production_ga_brp'] = aggregate_per_ga_and_brp(results['hourly_production_with_system_correction_and_grid_loss'])
+hourly_production_ga_brp = aggregate_per_ga_and_brp(results['hourly_production_with_system_correction_and_grid_loss'])
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:145]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102179 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 6]
```diff
@@ -87,86 +86,86 @@
...
 
 # STEP 16
-results['hourly_settled_consumption_ga_brp'] = aggregate_per_ga_and_brp(results['hourly_consumption_df'])
+hourly_settled_consumption_ga_brp = aggregate_per_ga_and_brp(results['hourly_consumption_df'])
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:148]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102179 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 7]
```diff
@@ -87,86 +86,86 @@
...
 
 # STEP 17
-results['flex_settled_consumption_ga_brp'] = aggregate_per_ga_and_brp(results['flex_consumption_with_grid_loss'])
+flex_settled_consumption_ga_brp = aggregate_per_ga_and_brp(flex_consumption_with_grid_loss)
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:151]: according to https://dev.azure.com/energinet/Datahub/_boards/board/t/Joules/Stories?workitem=102179 this should be added to the result

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 8]
```diff
@@ -87,86 +86,86 @@
...
 
-# Join additional data with added grid loss
-results['combined_grid_loss'] = combine_added_grid_loss_with_master_data(results['added_system_correction_df'], grid_loss_sys_cor_master_data_df)
+# Join additional data with added grid loss !
+results['combined_grid_loss'] = combine_added_grid_loss_with_master_data(added_system_correction_df, grid_loss_sys_cor_master_data_df)
 
-# STEP 10
-results['flex_consumption_with_grid_loss'] = adjust_flex_consumption(results['flex_consumption_df'],
-                                                                     results['added_grid_loss_df'],
+# STEP 10 !
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:125]: what does the `!` mean?

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 9]
```diff
@@ -87,86 +86,86 @@
...
+flex_consumption_with_grid_loss = adjust_flex_consumption(results['flex_consumption_df'],
+                                                                     added_grid_loss_df,
                                                                      grid_loss_sys_cor_master_data_df)
 
-# STEP 11
+# STEP 11 !
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:130]: what does the `!` mean?

**Review by mknic — CHANGES_REQUESTED** [review]

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 10]
```diff
@@ -87,86 +86,86 @@
 # create a keyvalue dictionary for use in postprocessing each result are stored as a keyval with value being dataframe
 results = {}
 
-# STEP 1
+# STEP 1 !
```
Comment (mknic) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:89]: I guess the `!` is to mark the steps being added to the result set? Maybe not that obvious though. Could we outline it in words instead maybe?

**Review by mknic — CHANGES_REQUESTED** [review]

File: source/databricks/aggregation-jobs/aggregation_trigger.py (hunk 10 above)
Comment (kristianschneider) [reply source/databricks/aggregation-jobs/aggregation_trigger.py:89]: Yes. Right now it was just for our own housekeeping

**Review by mknic — APPROVED** [review]

**Comment by ZavezX** [comment]
I think we should keep the strategy to save all results from our aggregations regardless of performance, as we would need to store those results anyway. If we experience performance issues, then let's create a new task to fix it in another way. I suggest the PR to be abandoned.

**Comment by kristianschneider** [comment]
@ZavezX Are you sure we need to store all steps ? That is not my impression

**Comment by mknic** [comment]
@kristianschneider not all steps are used in the data packages sent to the post office but as far as I understand the end-users need to be able to see the results for all steps for each completed process so we need to store each result at some point.

**Review by ZavezX — APPROVED** [review]

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 11]
```diff
@@ -103,44 +102,44 @@
 results['hourly_production_df'] = aggregate_hourly_production(df)
 
 # STEP 6
-results['grid_loss_df'] = calculate_grid_loss(results['net_exchange_per_ga_df'],
+grid_loss_df = calculate_grid_loss(results['net_exchange_per_ga_df'],
```
Comment (ZavezX) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:105]: I think this would be good to keep.

**Review by ZavezX — CHANGES_REQUESTED** [review]

File: source/databricks/aggregation-jobs/aggregation_trigger.py (hunk 11 above)
Comment (kristianschneider) [reply source/databricks/aggregation-jobs/aggregation_trigger.py]: Done

**Review by ZavezX — APPROVED** [review]

**Pull request merged**